libraries/bsg_manycore_uart_responder.cpp

```diff
@@ -3,36 +3,82 @@
 #include <bsg_manycore_printing.h>
 #include <stdio.h>
 
-#define UART_EPA 0xEADC
+enum hb_mc_uart_epa_indx {
+  STDOUT_EPA_INDX, 
+  STDERR_EPA_INDX, 
+  BRANCH_TRACE_EPA_INDX,
 
-static hb_mc_request_packet_id_t ids [] = {
-        RQST_ID( RQST_ID_ANY_X, RQST_ID_ANY_Y, RQST_ID_ADDR(0xEADC) ),
-        { /* sentinel */ },
+  HB_MC_NUM_UART_EPAS
 };
 
+static hb_mc_request_packet_id_t ids [HB_MC_NUM_UART_EPAS];
+
+static FILE* uart_streams [HB_MC_NUM_UART_EPAS];
+
 static int init(hb_mc_responder_t *responder,
-		hb_mc_manycore_t *mc)
+    hb_mc_manycore_t *mc)
 {
-	bsg_pr_dbg("hello from %s\n", __FILE__);
-	responder->responder_data = stdout;
+  bsg_pr_dbg("hello from %s\n", __FILE__);
+
+  for(int i=STDOUT_EPA_INDX; i<HB_MC_NUM_UART_EPAS; i++) {
```

Comment:
This work can be done at compile time like so:
```c
static hb_mc_request_packet_id_t ids [] = {
    [STDOUT_EPA_INDX]  = RQST_ID( RQST_ID_ANY_X, RQST_ID_ANY_Y, RQST_ID_ADDR(0xEADC),
    [STDERR_EPA_INDX]  = RQST_ID( RQST_ID_ANY_X, RQST_ID_ANY_Y, RQST_ID_ADDR(0xEEE0),
    [BRANCH_TRACE_EPA_INDX] = RQST_ID( RQST_ID_ANY_X, RQST_ID_ANY_Y, RQST_ID_ADDR(0xEEE4), 
};

static FILE * uart_streams [] = {
    [STDOUT_EPA_INDX] = stdout,
    [STDERR_EPA_INDX] = stderr,
    [BRANCH_TRACE_EPA_INDX] = stderr,
}; 
```
personally, I find that more readable than a finite for loop with an inner switch statement.

Reply:
Thanks! I used for loop as a way to use enums for ordering, I was not aware of this syntactic sugar, this is lot more elegant!